### PR TITLE
Enable make shell to work with finch in addition of docker

### LINKED ISSUE #539: Enable make shell to work with finch in addition of docker
### What would you like to be added?

I would like to use finch on macOS with `make shell` 
https://github.com/runfinch/finch

I was able to get it working with two changes
- Update hack/terraform context to use $HOME (finch can't mount /tmp)
```
TEMP="$HOME/tmp/eks-workshop-shell-env" 
```
- Update hack/shell.sh 
Replace (s/docker /finch /)
Update docker run with new TEMP location
Update `-e` flags to specify the value `
```
finch run --rm -it --env-file $HOME/tmp/eks-workshop-shell-env \
  -v $SCRIPT_DIR/../environment/workspace:/workspace \
  -e AWS_ACCESS_KEY_ID="$AWS_ACCESS_KEY_ID" -e AWS_SECRET_ACCESS_KEY="$AWS_SECRET_ACCESS_KEY" -e AWS_SESSION_TOKEN="$AWS_SESSION_TOKEN" \
  $container_image $shell_command
```

### Why is this needed?

I don't have docker desktop on macOS, only have finch installed.

### PR DESCRIPTION
**Pull request by csantanapr**
#### What this PR does / why we need it:

Enable use of `finch`

#### Which issue(s) this PR fixes:

Fixes #539

#### Quality checks

- [x] My content adheres to the style guidelines
- [x] I ran `make test` or `make e2e-test` and it was successful

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: hack/run-tests.sh [hunk 1]
```diff
@@ -23,16 +27,16 @@ echo "Building container images..."
...
 
 source $SCRIPT_DIR/lib/generate-aws-creds.sh
 
 echo "Running test suite..."
 
-docker run --rm --env-file /tmp/eks-workshop-shell-env \
+$CONTAINER_CLI run --rm --env-file ${TEMP:-/tmp}/eks-workshop-shell-env \
   -v $SCRIPT_DIR/../website/docs:/content \
   -v $SCRIPT_DIR/../environment/workspace:/workspace \
-  -e "AWS_ACCESS_KEY_ID" -e "AWS_SECRET_ACCESS_KEY" -e "AWS_SESSION_TOKEN" -e "AWS_DEFAULT_REGION" \
```
Comment (niallthomson) [inline hack/run-tests.sh:37]: Is this for readability?

File: hack/lib/terraform-context.sh [hunk 2]
```diff
@@ -2,6 +2,4 @@ terraform_dir="$SCRIPT_DIR/../$terraform_context"
 
 export ASSUME_ROLE=$(terraform -chdir=$terraform_dir output -raw iam_role_arn)
 
-TEMP='/tmp/eks-workshop-shell-env'
-
-terraform -chdir=$terraform_dir output -raw environment_variables > $TEMP
+terraform -chdir=$terraform_dir output -raw environment_variables > ${TEMP:-/tmp}/eks-workshop-shell-env
```
Comment (niallthomson) [inline hack/lib/terraform-context.sh:5]: Can you explain what drives the need for an explicit TEMP variable for finch specifically?

**Review by niallthomson — COMMENTED** [review]
Really appreciate this PR @csantanapr much needed. Add a couple of comments.

File: hack/lib/terraform-context.sh (hunk 2 above)
Comment (csantanapr) [reply hack/lib/terraform-context.sh]: I was using old version of finch and was not reading from /tmp, now is working

**Comment by niallthomson** [comment]
Looks good, thanks @csantanapr !

**Pull request merged**